### DISCUSSION TITLE
custom controller with dynamic data

### DISCUSSION THREAD

**Discussion by ghost** [strapi/strapi#6057] [Legacy Topics]
I am trying to create a custom controller which will be used as a webhook e.g. I will add the post link in GitHub app so that any update will be pushed to my custom controller but this payload has not defined any way we can pull this off in Strapi?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by ghost** [reply]
Thank you so much.

Although in issue template I selected the discussion type only don't know how this happened. Anyway thanks for moving it into the discussion.

**Comment by lauriejim** [comment]
Hello! To do that, you will have to create a custom API.

**Path —** `api/githubwebhook/config/routes.json`

```json
{
  "routes": [
    {
      "method": "POST",
      "path": "/githubwebhook",
      "handler": "Githubwebhook.manage",
      "config": {
        "policies": []
      }
    }
  ]
}
```

**Path —** `api/githubwebhook/controllers/Githubwebhook.js`

```js
module.exports = {
  async create(ctx) {
    const event = ctx.headers['x-github-event'];
    const signature = ctx.headers['x-hub-signature'];
    const delivery = ctx.headers['x-github-delivery'];
    const body = ctx.request.body;

    // Do you custom logic

   return 'Everything is awesome';
  }
};
```

**Reply by ghost** [reply]
How do I store this data in the database?
